src/desul/Macros.h

```diff
@@ -4,9 +4,9 @@
 #if defined(__CUDACC__) || defined(__HIPCC__)
```

Comment:
I've added what I think will work for SYCL. I'll need someone else with expertise in OpenMP target offload and OpenACC to figure out a good pattern for those programming models.

Reply:
I think all that logic exists in the atomics subfolder, perhaps we can promote it, or if we need something additional we have a pretty complete cascade for. ompiler and language detection in camp: https://github.com/trws/camp/blob/fix_ci/include/camp/defines.hpp

Reply:
So on that note, it seems as if the atomics were really designed as a standalone library rather than as part of a set of shared components. What is the overall layout we want for this library if we start adding more components? I can see two potential approaches:

1. The current approach: have a main folder for each grouping of components (atomics, containers, math, algorithms, etc...). The main difficulty is deciding how to group components and dealing with components that reach across groups.
2. Have a <repo root>/src/desul directory that contains the public headers, and then internal implementation details are put in subdirectories in that main directory as needed.

Reply:
My original hope was that we would go for something somewhat between the two.

* Each main folder contains a component
* We'll want to factor out a "common" component for things like determining the compiler and so-forth I think as well so we don't need to reach between them for unrelated things
* At the top level, have a build that builds all the components, and offers a desul include directory with a `desul.hpp` that includes all the components.  While the mega-header approach isn't very nice for headers, it's _very_ nice when we start talking about modules, so planning ahead just a bit.